cmd/check/main.go

```diff
@@ -32,12 +33,27 @@ func main() {
 		log.Fatalf("failed unmarshalling: %s", err)
 	}
 
-	client := boshio.NewClient(nil, nil)
+	client := boshio.NewClient(nil, nil, checkRequest.Source.ForceRegular)
 	stemcells, err := client.GetStemcells(checkRequest.Source.Name)
 	if err != nil {
 		log.Fatalf("failed getting stemcell: %s", err)
 	}
 
+	supportsLight := client.SupportsLight(stemcells)
+	if supportsLight {
+		if checkRequest.Source.ForceRegular {
+			regularFilter := func(stemcell boshio.Stemcell) bool {
+				return stemcell.Regular != nil
+			}
+			stemcells = client.FilterStemcells(regularFilter, stemcells)
+		} else {
+			lightFilter := func(stemcell boshio.Stemcell) bool {
+				return stemcell.Light != nil
```

Comment:
I'd actually like to request that we change this. Instead of the complexity of having to call what is a filtering function a `lamba` can't we just move all of the filtering logic into the function that is actually there to filter stemcells? To me, it reduces the complexity of the main.go and pushes the responsibility down to the package that should be holding onto it.